File: Csharp/CodeWars/src/Resolutions/MaximumLengthDifference.cs
namespace CodeWars.Resolutions;

public class MaximumLengthDifference
{
    public static int Mxdiflg(string[] a1, string[] a2)
    {
        if (a1.Length == 0 || a2.Length == 0) return -1;

        int[] a1Lengths = [.. a1.Select((s) => s.Length)];
        int[] a2Lengths = [.. a2.Select((s) => s.Length)];

        int maxA1 = a1Lengths.Max();
        int minA1 = a1Lengths.Min();
        int maxA2 = a2Lengths.Max();
        int minA2 = a2Lengths.Min();

        return Math.Max(Math.Abs(maxA1 - minA2), Math.Abs(maxA2 - minA1));
    }
}

File: TypeScript/Codewars/src/resolutions/Maximum-Length-Difference.ts
function mxdiflg(a1: string[], a2: string[]): number {
	if (a1.length === 0 || a2.length === 0) return -1;

	const a1Lengths = a1.map((s) => s.length);
	const a2Lengths = a2.map((s) => s.length);

	const maxA1 = Math.max(...a1Lengths);
	const minA1 = Math.min(...a1Lengths);
	const maxA2 = Math.max(...a2Lengths);
	const minA2 = Math.min(...a2Lengths);

	return Math.max(Math.abs(maxA1 - minA2), Math.abs(maxA2 - minA1));
}

console.log(
	mxdiflg(["ejjjjmmtthh", "zxxuueeg", "aanlljrrrxx", "dqqqaaabbb", "oocccffuucccjjjkkkjyyyeehh"], ["bbbaaayddqbbrrrv"]),
);
